### ISSUE TITLE
[3] Upgrade exoscale terraform provider

### ISSUE THREAD

**Issue by Xartos** [elastisys/compliantkubernetes-kubespray#270]
Currently the terraform script is based on the version [0.21](https://github.com/kubernetes-sigs/kubespray/blob/master/contrib/terraform/exoscale/modules/kubernetes-cluster/versions.tf#L5) and hasn't really been touched since then.
The latest version as of writing is [0.46](https://github.com/exoscale/terraform-provider-exoscale/blob/master/CHANGELOG.md#0460-february-23-2023). They have for example removed the `exoscale_compute` resource that we are using in favor of `exoscale_compute_instance` to support their new flavors and some changes like that.

We should probably try to fix this to not lag behind to much and stuff starts to break.

## Definition of Done

- [ ] The exoscale provider in the kubespray terraform scripts for exoscale is updated.

**Comment by OlleLarsson** [comment]
[0.47 released](https://github.com/exoscale/terraform-provider-exoscale/blob/master/CHANGELOG.md#0470-april-24-2023).

**Comment by crssnd** [comment]
also try to fix [this issue](https://github.com/elastisys/compliantkubernetes-kubespray/issues/241) at the same time

**Comment by OlleLarsson** [comment]
Move back to prio board?

**Comment by crssnd** [comment]
no update from the upstream so we'll move it to the backlog

**Comment by OlleLarsson** [comment]
@Ajarmar @Pavan-Gunda can we close this?

**Comment by Pavan-Gunda** [comment]
Yes, Since the roadmap item is a "won't do". I think we can close this and the parent epic.
